File: java/Password.java
import java.security.SecureRandom;

public class Password {
    public Password() {
        System.out.println("This is a class for creating passwords. Example: Password.getpassword(10). This creates\na password with length 10. If you want to print it, write System.out.println(Password.getpassword(10)).");
    }

    private static final String CHARACTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+[]{}|;:,.<>?";

    public static String generatePassword(int length) {
        SecureRandom random = new SecureRandom();
        StringBuilder password = new StringBuilder(length);

        for (int i = 0; i < length; i++) {
            int randomIndex = random.nextInt(CHARACTERS.length());
            password.append(CHARACTERS.charAt(randomIndex));
        }

        return password.toString();
    }

    public static String getpassword(int length) {
        return generatePassword(length);
    }

    public static void main(String[] args) {
        Password passwordGenerator = new Password();

        int length = 10;
        String generatedPassword = passwordGenerator.getpassword(length);
        System.out.println("Generated Password: " + generatedPassword);
    }
}

File: c++/Password.cpp
#include <iostream>
#include <random>
#include <string>

class Password {
public:
    Password() {
        std::cout << "This is a class for creating passwords. Example: Password::getpassword(10). This creates\na password with length 10. If you want to print it, write std::cout << Password::getpassword(10) << std::endl;";
    }

    static std::string generatePassword(int length) {
        static const std::string CHARACTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()-_=+[]{}|;:,.<>?";
        std::string password;
        std::random_device rd;
        std::mt19937 generator(rd());
        std::uniform_int_distribution<int> distribution(0, CHARACTERS.size() - 1);

        for (int i = 0; i < length; ++i) {
            password += CHARACTERS[distribution(generator)];
        }

        return password;
    }

    static std::string getpassword(int length) {
        return generatePassword(length);
    }
};

int main() {
    Password passwordGenerator;

    int length = 10;
    std::string generatedPassword = passwordGenerator.getpassword(length);
    std::cout << "Generated Password: " << generatedPassword << std::endl;

    return 0;
}
